### DISCUSSION TITLE
Run job only on the instance that created it?

### DISCUSSION THREAD

**Discussion by dmurphy-ambra** [jobrunr/jobrunr#349] [Q&A] [answered]
In our environment, some aspects are clustered and shared across our instances, but some aspects are not.  In those cases, we would like to schedule or enqueue a job, but to make sure that it only runs on the instance which created it.  The reason we need this feature is because we have temporary files that are local to an instance in certain situations, and therefore only the instance with access to those temp files can execute the job.

Is there a way to handle this with the existing JobRunr or JobRunr Pro? In JobRunr Pro, there are server tags, which allow you to classify which server to run certain jobs on, but does not seem like that solves this particular use case.

**Comment by rdehuyss** [comment, marked as answer]
See #350.

This will be possible in JobRunr 5.0 thanks to the following syntax:

```java
    @Job(runOnServerWithTag = "%CURRENT_SERVER")
    public void doWorkOnCurrentServerThatMatches() {...}
```
